lib/romeo/auth.ex

```diff
@@ -44,8 +44,16 @@ defmodule Romeo.Auth do
 
 
   defp do_authenticate(mechanism, conn) do
-    Logger.info fn -> "Authenticating with #{mechanism}" end
-    authenticate_with(mechanism, conn)
+    {:ok, conn} =
+      case mechanism do
+        {name, mechanism_fn} ->
```

Comment:
Let's change this so that people can provide a module that implements `authenticate/1`.

```elixir
{"X-OAUTH2", XOAUTH2}
```

```elixir
defmodule XOAUTH2 do
  def authenticate(%{transport: mod, jid: jid, password: password} = conn) do
    payload = <<0>> <> jid <> <<0>> <> password
    additional_attrs = [{"auth:service", "oauth2"}, {"xmlns:auth", "http://www.google.com/talk/protocol/auth"}]
    mod.send(conn, Romeo.Stanza.auth("X-OAUTH2", Romeo.Stanza.base64_cdata(payload), additional_attrs))
  end
end
```

This would allow others to publish a hex package to provide auth extensions following this simple interface.